main.go

```diff
@@ -29,23 +30,60 @@ func main() {
 	}
 	source := gopacket.NewPacketSource(src, dec)
 
-	var eth layers.Ethernet
-	var ip4 layers.IPv4
-	var ip6 layers.IPv6
-	var udp layers.UDP
-	var tag layers.Dot1Q
-	parser := gopacket.NewDecodingLayerParser(layers.LayerTypeEthernet, &tag, &eth, &ip4, &ip6, &udp)
-	decoded := []gopacket.LayerType{}
-
 	for packet := range source.Packets() {
-		parser.DecodeLayers(packet.Data(), &decoded)
+		// // Extract DOT1Q tag
+		// var tag *layers.Dot1Q
+		// if parsedTag := packet.Layer(layers.LayerTypeDot1Q); parsedTag != nil {
+		// 	tag, _ = parsedTag.(*layers.Dot1Q)
+		// }
+
+		var ip4 layers.IPv4
+		if parsedIP := packet.Layer(layers.LayerTypeIPv4); parsedIP != nil {
+			ip4 = *parsedIP.(*layers.IPv4)
+		}
+
+		var ip6 layers.IPv6
+		if parsedIP := packet.Layer(layers.LayerTypeIPv6); parsedIP != nil {
+			ip6 = *parsedIP.(*layers.IPv6)
+		}
+
+		var udp layers.UDP
+		if parsedUDP := packet.Layer(layers.LayerTypeUDP); parsedUDP != nil {
+			udp = *parsedUDP.(*layers.UDP)
+		}
+
 		// Detect Bonjour packets
 		if ip4.DstIP.String() == "224.0.0.251" || ip6.DstIP.String() == "ff02::fb" {
 			if udp.DstPort == 5353 {
 				// Print time for logging / debugging purposes
 				fmt.Printf("[%v] New Bonjour packet detected from %v\n",
-					time.Now().Format("02/01/2006 15:04:05"), ip4.SrcIP) // Custom time layouts must use the reference time: Mon Jan 2 15:04:05 MST 2006
+					time.Now().Format("02/01/2006 15:04:05"), ip4.SrcIP.String()) // Custom time layouts must use the reference time: Mon Jan 2 15:04:05 MST 2006
```

Comment:
You should check if ip4.SrcIP or ip6.SrcIP is not nil, and print the right one accordingly (the request might come from an IPv6 address, but it will be shown as nil).